src/tir/transforms/lower_tvm_builtin.cc

```diff
@@ -48,30 +48,54 @@ inline PrimExpr StackAlloca(std::string type, size_t num) {
 // These information are needed during codegen.
 class BuiltinLower : public StmtExprMutator {
  public:
-  Stmt Build(Stmt stmt) {
-    stack_shape_ = Var("stack_shape", DataType::Handle());
-    stack_array_ = Var("stack_array", DataType::Handle());
-    stack_value_ = Var("stack_value", DataType::Handle());
-    stack_tcode_ = Var("stack_tcode", DataType::Handle());
+  // Record stack frame for existing scope.
+  struct AllocaScope {
+    Var stack_shape_ = Var("stack_shape", DataType::Handle());
+    Var stack_array_ = Var("stack_array", DataType::Handle());
+    Var stack_value_ = Var("stack_value", DataType::Handle());
+    Var stack_tcode_ = Var("stack_tcode", DataType::Handle());
+
+    int64_t max_shape_stack_{-1};
+    uint64_t max_array_stack_{0};
+    uint64_t max_arg_stack_{0};
+
+    int64_t run_shape_stack_{-1};
+    uint64_t run_array_stack_{0};
+    uint64_t run_arg_stack_{0};
+  };
+
+  Stmt Build(Stmt stmt) { return this->RealizeAlloca(stmt); }
+
+  // Allcoate stack frames, only at parallel-for or root.
+  Stmt RealizeAlloca(Stmt stmt) {
```

Comment:
`VisitBodyAndRealizeAlloca(Stmt body)`
for readability

Reply:
Done.